source/common/event/timer_impl.cc

```diff
@@ -17,16 +17,28 @@ void TimerUtils::millisecondsToTimeval(const std::chrono::milliseconds& d, timev
   tv.tv_usec = usecs.count();
 }
 
-TimerImpl::TimerImpl(Libevent::BasePtr& libevent, TimerCb cb) : cb_(cb) {
+TimerImpl::TimerImpl(Libevent::BasePtr& libevent, TimerCb cb, Dispatcher& dispatcher)
+    : cb_(cb), dispatcher_(dispatcher) {
   ASSERT(cb_);
   evtimer_assign(
       &raw_event_, libevent.get(),
-      [](evutil_socket_t, short, void* arg) -> void { static_cast<TimerImpl*>(arg)->cb_(); }, this);
+      [](evutil_socket_t, short, void* arg) -> void {
+        TimerImpl* timer = static_cast<TimerImpl*>(arg);
+        if (timer->object_ == nullptr) {
+          timer->cb_();
+          return;
+        }
+        ScopeTrackerScopeState scope(timer->object_, timer->dispatcher_);
```

Comment:
This change looks good to me, but do we have any tests that explicitly cover this? I might be missing it in the large test change.

Reply:
No, I totally lost it during the rebase, thanks for catching that.
Thankfully where my local state was destroyed, github saved it for me :-)
https://github.com/envoyproxy/envoy/commit/ce642b2e501e92b53d5d49acb77c8696d514d721